Making progress:

```
~ make stress TAGS=deadlock PKG=./storage TESTS=TestRefreshPendingCommands STRESSFLAGS="-maxfails 1 -stderr -p 16"
...
    --- FAIL: TestRefreshPendingCommands/#00 (2.34s)
    	client_raft_test.go:1000: range 1: replica {3 3 3} not lease holder; current lease is repl={3 3 3} start=0.900000123,17 exp=2.700000125,42 pro=1.800000125,43
FAIL


ERROR: exit status 1

7310 runs completed, 1 failures, over 20m31s
```

Before this PR the above `stress` invocation would fail within a minute or two.

@spencerkimball, @andreimatei How could the above error occur?

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -967,17 +967,25 @@ func TestRefreshPendingCommands(t *testing.T) {
 			mtc.stopStore(0)
 			mtc.restartStore(0)
 
+			// Disable node liveness heartbeats which can reacquire leases when we're
+			// trying to expire them. We pause liveness heartbeats here after node 0
+			// was restarted (which creates a new NodeLiveness).
+			pauseNodeLivenessHeartbeats(mtc, true)
+
 			// Expire existing leases (i.e. move the clock forward, but don't
 			// increment epochs). This allows node 2 to grab the lease later
 			// in the test.
-			mtc.expireLeasesWithoutIncrementingEpochs()
-			// Drain leases from nodes 0 and 1 to prevent them from grabbing any new
-			// leases.
-			for i := 0; i < 2; i++ {
-				if err := mtc.stores[i].DrainLeases(true); err != nil {
-					t.Fatalf("store %d: %v", i, err)
+			testutils.SucceedsSoon(t, func() error {
+				mtc.expireLeasesWithoutIncrementingEpochs()
+				// Drain leases from nodes 0 and 1 to prevent them from grabbing any new
+				// leases.
+				for i := 0; i < 2; i++ {
+					if err := mtc.stores[i].DrainLeases(true); err != nil {
+						return errors.Wrapf(err, "store %d", i)
+					}
 				}
-			}
+				return nil
+			})
 
 			// Restart node 2 and wait for the snapshot to be applied. Note that
 			// waitForValues reads directly from the engine and thus isn't executing

